src/autorotate-iio.cpp

```diff
@@ -71,16 +71,24 @@ class WayfireAutorotateIIO : public wf::plugin_interface_t
     wf_option rotate_down_opt;
 
     activator_callback on_rotate_left = [=] (wf_activator_source src, int32_t) {
-        on_rotate_binding(WL_OUTPUT_TRANSFORM_270);
+        if (!output->can_activate_plugin(grab_interface))
+            return false;
+        return on_rotate_binding(WL_OUTPUT_TRANSFORM_270);
     };
     activator_callback on_rotate_right = [=] (wf_activator_source src, int32_t) {
-        on_rotate_binding(WL_OUTPUT_TRANSFORM_90);
+        if (!output->can_activate_plugin(grab_interface))
+            return false;
+        return on_rotate_binding(WL_OUTPUT_TRANSFORM_90);
     };
     activator_callback on_rotate_up = [=] (wf_activator_source src, int32_t) {
-        on_rotate_binding(WL_OUTPUT_TRANSFORM_NORMAL);
+        if (!output->can_activate_plugin(grab_interface))
+            return false;
+        return on_rotate_binding(WL_OUTPUT_TRANSFORM_NORMAL);
     };
     activator_callback on_rotate_down = [=] (wf_activator_source src, int32_t) {
-        on_rotate_binding(WL_OUTPUT_TRANSFORM_180);
+        if (!output->can_activate_plugin(grab_interface))
```

Comment:
I'd move this check in `on_rotate_binding`, this way you don't need to write it each time.

Reply:
I don't understand what you're saying. AFAIU, it needs this so a binding bound to this will still work in wcm keybinding grabs.

Reply:
What I mean is that you are doing the same thing in all callbacks: `if (!can_activate) return false; else return on_rotate_binding()` Instead, you can have the `if(!can_activate)` check inside `on_rotate_binding()` (and this way you don't have to type `if (!can_activate)` each time).

Reply:
Ah, I see what you mean.